xcode/Subconscious/Shared/Services/UserProfileService.swift

```diff
@@ -112,6 +118,11 @@ struct UserProfileCacheEntry {
     let version: Cid
 }
 
+struct UserProfileCacheKey: Hashable {
+    let did: Did
+    let petname: Petname?
+}
```

Comment:
The same user with the same `Did` can have different `UserProfile`s depending on the context you encounter them.

A trivial example: following the same `Did` under two `Petname`s changes which name is chosen as the "primary" name, depending on which version you're looking at.

Reply:
Worth saying: I'm open to the idea of dropping the in-memory cache. The main drawback is that we'd be sending a query to SQLite for each post in the feed except in the case of _our_ profile which is read from our sphere.

Reply:
Yep! I was considering that in this comment: https://github.com/subconsciousnetwork/subconscious/pull/925#discussion_r1351439703

**But** I realised today that since `EntryStub` now had the `Did` attached, I don't need to load a profile at all for the feed. In fact, I managed to drop `author` from `EntryStub` entirely (I wasn't a fan of having to introduce it in the first place).

Now, we can produce the feed + backlinks + transcludes without having to fetch the user's profile in the process. _If_ we introduce something like customisable PFPs in the future we will have to revisit this but I would argue that that's a whole seperate domain.

As a result, there's really no benefit coming from the in-memory cache so... I removed it. One less invalidation layer to care about.